core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/AddProvisionedResourceCommandHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.transfer.core.transfer;
+
+import org.eclipse.dataspaceconnector.spi.command.CommandHandler;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferProcessManager;
+import org.eclipse.dataspaceconnector.transfer.core.command.commands.AddProvisionedResourceCommand;
+
+import java.util.List;
+
+/**
+ * Processes a {@link AddProvisionedResourceCommand} by delegating it to the {@link TransferProcessManager}.
+ *
+ * This class exists to avoid coupling the TPM to the command handler registry.
+ */
+public class AddProvisionedResourceCommandHandler implements CommandHandler<AddProvisionedResourceCommand> {
```

Comment:
Any reason why this would **not** extend `SingleTransferProcessCommandHandler`? 
Then, the `modify` operation would receive the correct `TransferProcess`, and we would not have to store an `TransferProcessManagerImpl` instance. Then, the only operation necessary would be to add the provisioned resource, returning `true` signals the base class to actually invoke `update`: 
```

@Override
public void handle(SingleTransferProcessCommand command){
   this.provisionResult = ((AddProvisionedResourceCommand) command).getProvisionResponse()
}

@Override
protected boolean modify(TransferProcess process) {
    process.addProvisionedResource(this.provisionResult/*...*/);
    return true;
}
```